cmake/CMakeLists.txt

```diff
@@ -0,0 +1,160 @@
+check_function_exists(eventfd                   HAVE_EVENTFD)
+check_function_exists(pipe2                     HAVE_PIPE2)
+check_function_exists(syslog                    HAVE_SYSLOG)
+
+check_include_files(asm/types.h      HAVE_ASM_TYPES_H) 
+check_include_files(sys/eventfd.h    HAVE_EVENTFD) 
+check_include_files(dlfcn.h          HAVE_DLFCN_H)
+check_include_files(inttypes.h       HAVE_INTTYPES_H)
+check_include_files(stdint.h         HAVE_STDINT_H)
+check_include_files(stdio.h          HAVE_STDIO_H)
+check_include_files(stdlib.h         HAVE_STDLIB_H)
+check_include_files(string.h         HAVE_STRING_H)
+check_include_files(strings.h        HAVE_STRINGS_H)
+check_include_files(sys/stat.h       HAVE_SYS_STAT_H)
+check_include_files(sys/time.h       HAVE_SYS_TIME_H)
+check_include_files(sys/types.h      HAVE_SYS_TYPES_H)
+check_include_files(unistd.h         HAVE_UNISTD_H)
+check_include_files("stdlib.h;stdarg.h;string.h;float.h" STDC_HEADERS)
+
+check_symbol_exists(EFD_CLOEXEC     "sys/eventfd.h" HAVE_DECL_EFD_CLOEXEC)
+check_symbol_exists(EFD_NONBLOCK    "sys/eventfd.h" HAVE_DECL_EFD_NONBLOCK)
+check_symbol_exists(TFD_CLOEXEC     "sys/eventfd.h" HAVE_DECL_TFD_CLOEXEC)
+check_symbol_exists(timerfd_create  "sys/timerfd.h" HAVE_TIMERFD) 
+
+check_struct_has_member("struct timespec" tv_sec time.h HAVE_STRUCT_TIMESPEC)
+
+if(BUILD_SHARED_LIBS)
+  set(libusb_BUILD_SHARED_LIBS_DEFAULT ON)
+else (BUILD_SHARED_LIBS)
+  set(libusb_BUILD_SHARED_LIBS_DEFAULT OFF)
+endif (BUILD_SHARED_LIBS)
+
+option(libusb_BUILD_SHARED_LIBS "Build Shared Libraries for libusb" ${libusb_BUILD_SHARED_LIBS_DEFAULT})
+option(libusb_BUILD_TESTING "Build Tests" ON)
+option(ENABLE_LOGGING "Enable Logging" ON)
+
+if(WIN32)
+    set(PLATFORM_WINDOWS 1 CACHE INTERNAL "")
+endif()
+if(UNIX)
+    set(PLATFORM_POSIX 1  CACHE INTERNAL "")
+endif()
+
+if(("${CMAKE_CXX_COMPILER_ID}" STREQUAL Clang) OR ("${CMAKE_CXX_COMPILER_ID}" STREQUAL GNU))
+    check_c_compiler_flag("-fvisibility=hidden" HAVE_VISIBILITY)
+endif()
+
+if(HAVE_VISIBILITY)
+    set(DEFAULT_VISIBILITY "__attribute__((visibility(\"default\")))"  CACHE INTERNAL "")
+else()
+    set(DEFAULT_VISIBILITY ""  CACHE INTERNAL "")
+endif()
+
+configure_file(config.h.in config.h @ONLY)
+
+set(libusb_ROOT ../libusb)
+
+set(common_sources
+    "${libusb_ROOT}/core.c"
+    "${libusb_ROOT}/descriptor.c"
+    "${libusb_ROOT}/hotplug.c"
+    "${libusb_ROOT}/io.c"
+    "${libusb_ROOT}/sync.c"
+    "${libusb_ROOT}/strerror.c"
+
+    "${libusb_ROOT}/libusb.h"
+    "${libusb_ROOT}/libusbi.h"
+    "${libusb_ROOT}/version_nano.h"
+    "${libusb_ROOT}/version.h"
+
+    "${CMAKE_CURRENT_BINARY_DIR}/config.h"
+)
+
+if(libusb_BUILD_SHARED_LIBS)
+    add_library(usb-1.0 SHARED ${common_sources})
+else()
+    add_library(usb-1.0 STATIC ${common_sources})
+endif()
+
+target_include_directories(usb-1.0 PUBLIC "${libusb_ROOT}")
+target_include_directories(usb-1.0 PUBLIC "${CMAKE_CURRENT_BINARY_DIR}")
```

Comment:
`CMAKE_CURRENT_BINARY_DIR` may contain a lot of intermediate artifacts
maybe generate `config.h.in` into `gen_include/config.h`?
so we could can narrow the include path to a subfolder: `${CMAKE_CURRENT_BINARY_DIR}/gen_include`